demo/full/scripts/controllers/ControlBar.jsx

```diff
@@ -1,87 +1,188 @@
+     * - If current position is too far from live edge,
+     * then we should seek near to live edge.
+     * - If not, changed playback rate in order to move back
+     * close to the live.
+     */
+    const catchUp = () => {
+      if (liveGap > 10) {
+        player.dispatch("SEEK", maximumPosition - 5);
+      } else {
+        const factor = (liveGap - 5) / 4;
+        const rate = Math.round(
+          (liveGap > 5 ? Math.min(10, 1.1 + factor) : 1) * 10
+        ) / 10;
+        if (rate !== playbackRate) {
+          this.setState({ isCatchingUp: true });
+          player.dispatch("SET_PLAYBACK_RATE", rate);
+        }
+      }
+    };
+
+    if (player) {
+      const shouldCatchUp = isContentLoaded &&
+                            (liveGap > 7 || isCatchingUp && liveGap > 5) &&
+                            isStickingToTheLiveEdge &&
+                            lowLatencyMode;
+      if (shouldCatchUp) {
+        catchUp();
+      } else if (isCatchingUp) {
+        this.setState({ isCatchingUp: false });
+        player.dispatch("SET_PLAYBACK_RATE", 1);
+      }
+    }
+
+    const positionElement = (() => {
+      if (!isContentLoaded) {
+        return null;
+      } else if (isLive) {
+        return <LivePosition />;
+      } else {
+        return <PositionInfos
+          position={currentTime}
+          duration={duration}
+        />;
+      }
+    })();
+
+    return (
+      <div className="controls-bar-container">
+        <Progressbar
           player={player}
+          onSeek={() => changeStickToLiveEdge(false)}
         />
-        <Button
-          className={"control-button"}
-          onClick={stopVideo}
-          value={String.fromCharCode(0xf04d)}
-          disabled={isStopped}
-        />
-        { positionElement }
-        <div className="controls-right-side">
+        <div className="controls-bar">
+          <PlayPauseButton
+            className={"control-button"}
+            player={player}
+          />
           <Button
-            disabled={!isContentLoaded}
-            className='control-button'
-            onClick={toggleSettings}
-            value={String.fromCharCode(0xf013)}
+            className={"control-button"}
+            onClick={stopVideo}
+            value={String.fromCharCode(0xf04d)}
+            disabled={isStopped}
           />
-          <div className="volume">
-            <VolumeButton
-              className="control-button"
-              player={player}
+          <StickToLiveEdgeButton
+            isContentLoaded={isContentLoaded}
+            lowLatencyMode={lowLatencyMode}
+            isStickingToTheLiveEdge={isStickingToTheLiveEdge}
+            changeStickToLiveEdge={changeStickToLiveEdge}
+          />
+          {positionElement}
+          {isContentLoaded ?
+            <Button
+              className={"dot" + ((isCloseToLive && !isCatchingUp) ? " live" : "")}
+              onClick={() => {
+                if (isCatchingUp || !isCloseToLive) {
```

Comment:
So that's the exact opposite of what is computed in className.


It could be nicer to have a variable which tell what we're checking here (`shouldEnableLiveDot` or `isAtLiveEdge`...?)